libminifi/test/aws-tests/FetchS3ObjectTests.cpp

```diff
@@ -0,0 +1,174 @@
+  REQUIRE(LogTestController::getInstance().contains("key:s3.sseAlgorithm value:" + S3_SSEALGORITHM_STR));
+  REQUIRE(LogTestController::getInstance().contains("key:s3.version value:" + S3_VERSION));
+  std::ifstream s3_file(output_dir + "/" + INPUT_FILENAME);
+  std::string output((std::istreambuf_iterator<char>(s3_file)), std::istreambuf_iterator<char>());
+  REQUIRE(output == S3_CONTENT);
+  REQUIRE(mock_s3_wrapper_ptr->get_object_request.GetVersionId().empty());
+  REQUIRE(!mock_s3_wrapper_ptr->get_object_request.VersionIdHasBeenSet());
+  REQUIRE(mock_s3_wrapper_ptr->get_object_request.GetRequestPayer() == Aws::S3::Model::RequestPayer::NOT_SET);
+}
+
+TEST_CASE_METHOD(FetchS3ObjectTestsFixture, "Test empty optional S3 results", "[awsS3Config]") {
+  setRequiredProperties();
+  mock_s3_wrapper_ptr->returnEmptyS3Result();
+  test_controller.runSession(plan, true);
+  REQUIRE(LogTestController::getInstance().contains("key:s3.bucket value:" + S3_BUCKET));
+  REQUIRE(LogTestController::getInstance().contains("key:filename value:" + INPUT_FILENAME));
+  REQUIRE(LogTestController::getInstance().contains("key:path value:\n"));
+  REQUIRE(LogTestController::getInstance().contains("key:absolute.path value:" + INPUT_FILENAME));
+  REQUIRE(!LogTestController::getInstance().contains("key:mime.type", std::chrono::seconds(0), std::chrono::milliseconds(0)));
+  REQUIRE(!LogTestController::getInstance().contains("key:s3.etag", std::chrono::seconds(0), std::chrono::milliseconds(0)));
+  REQUIRE(!LogTestController::getInstance().contains("key:s3.expirationTime", std::chrono::seconds(0), std::chrono::milliseconds(0)));
+  REQUIRE(!LogTestController::getInstance().contains("key:s3.expirationTimeRuleId", std::chrono::seconds(0), std::chrono::milliseconds(0)));
+  REQUIRE(!LogTestController::getInstance().contains("key:s3.sseAlgorithm", std::chrono::seconds(0), std::chrono::milliseconds(0)));
+  REQUIRE(!LogTestController::getInstance().contains("key:s3.version", std::chrono::seconds(0), std::chrono::milliseconds(0)));
+  std::ifstream s3_file(output_dir + "/" + INPUT_FILENAME);
+  std::string output((std::istreambuf_iterator<char>(s3_file)), std::istreambuf_iterator<char>());
+  REQUIRE(output.empty());
+}
+
+TEST_CASE_METHOD(FetchS3ObjectTestsFixture, "Test subdirectories on AWS", "[awsS3Config]") {
+  setRequiredProperties();
+  plan->setProperty(s3_processor, "Object Key", "dir1/dir2/logs.txt");
+  test_controller.runSession(plan, true);
+  REQUIRE(LogTestController::getInstance().contains("key:filename value:logs.txt"));
+  REQUIRE(LogTestController::getInstance().contains("key:path value:dir1/dir2"));
+  REQUIRE(LogTestController::getInstance().contains("key:absolute.path value:dir1/dir2/logs.txt"));
+  std::ifstream s3_file(output_dir + "/" + INPUT_FILENAME);
+  std::string output((std::istreambuf_iterator<char>(s3_file)), std::istreambuf_iterator<char>());
+  REQUIRE(output.empty());
```

Comment:
Probably a good candidate to extract to FileUtils.h

Reply:
Extraced in [6097258](https://github.com/apache/nifi-minifi-cpp/pull/970/commits/60972586e82e9e7786adf01ce84c305e88dc7224)